app/src/main/java/com/naman14/algovisualizer/algorithm/search/LinearSearch.java

```diff
@@ -0,0 +1,109 @@
+package com.naman14.algovisualizer.algorithm.search;
+
+import android.app.Activity;
+
+import com.naman14.algovisualizer.LogFragment;
+import com.naman14.algovisualizer.algorithm.Algorithm;
+import com.naman14.algovisualizer.algorithm.DataHandler;
+import com.naman14.algovisualizer.visualizer.BinarySearchVisualizer;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+/**
+ * Created by mayank on 27/11/16.
+ */
+
+public class LinearSearch extends Algorithm implements DataHandler {
+
+    private BinarySearchVisualizer visualizer;
+    private int[] array;
+
+    private List<Integer> positions = new ArrayList<>();
+
+    public LinearSearch(BinarySearchVisualizer visualizer, Activity activity, LogFragment logFragment) {
+        this.visualizer = visualizer;
+        this.activity = activity;
+        this.logFragment = logFragment;
+    }
+
+    public void setData(final int[] array) {
+        activity.runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                visualizer.setData(array);
+            }
+        });
+        start();
+        prepareHandler(this);
+        sendData(array);
+    }
+
+    private void search() {
+
+        logArray("Array - ", array);
+
+        int rnd = new Random().nextInt(array.length);
+        int data = array[rnd];
+        addLog("Searching for " + data);
+
+        int n = array.length - 1;
+
+        for (int i = 0 ; i<n ; i++) {
+
+            highlight(i, i);
```

Comment:
A better visualization would be to highlight the already searched values (`highlight(0,i)`) and `highlightTrace(i)` the position being currently searched